### PR TITLE
Reback shimv2 tests on rhel, sles and opensuse 

### PR DESCRIPTION
**Pull request by lifupan**
We can disable seccomp support when building runc on those OS, thus
we can reback the shimv2 tests.

### CODE REVIEW

**Comment by chavafg** [comment]
opensuse package btrfs-progs-devel not found, seems like it has another name?

```
07:50:35 No provider of 'btrfs-progs-devel' found.
07:50:35 Build step 'Execute shell' marked build as failure
```

**Comment by chavafg** [comment]
searching for the package name on a opensuse container found:
```
# zypper search btrfs
Loading repository data...
Reading installed packages...

S | Name                  | Summary                                                         | Type      
--+-----------------------+-----------------------------------------------------------------+-----------
  | btrfs-extent-same     | Debug/Test tool to exercise the btrfs out-of-band deduplicati-> | package   
  | btrfsmaintenance      | Scripts for btrfs periodic maintenance tasks                    | package   
  | btrfsprogs            | Utilities for the Btrfs filesystem                              | package   
  | btrfsprogs            | Utilities for the Btrfs filesystem                              | srcpackage
  | btrfsprogs-udev-rules | Udev rules for configuring btrfs file systems                   | package   
  | ghc-btrfs             | Bindings to the btrfs API                                       | package   
  | ghc-btrfs-devel       | Haskell btrfs library development files                         | package   
  | libbtrfs-devel        | Include Files and Libraries for developing with Btrfs           | package   
  | libbtrfs0             | Library for interacting with Btrfs                              | package  
```

So I suppose the package we're looking for is `libbtrfs-devel `?
/cc @marcov @GabyCT

**Comment by lifupan** [comment]
> searching for the package name on a opensuse container found:
> 
> ```
> # zypper search btrfs
> Loading repository data...
> Reading installed packages...
> 
> S | Name                  | Summary                                                         | Type      
> --+-----------------------+-----------------------------------------------------------------+-----------
>   | btrfs-extent-same     | Debug/Test tool to exercise the btrfs out-of-band deduplicati-> | package   
>   | btrfsmaintenance      | Scripts for btrfs periodic maintenance tasks                    | package   
>   | btrfsprogs            | Utilities for the Btrfs filesystem                              | package   
>   | btrfsprogs            | Utilities for the Btrfs filesystem                              | srcpackage
>   | btrfsprogs-udev-rules | Udev rules for configuring btrfs file systems                   | package   
>   | ghc-btrfs             | Bindings to the btrfs API                                       | package   
>   | ghc-btrfs-devel       | Haskell btrfs library development files                         | package   
>   | libbtrfs-devel        | Include Files and Libraries for developing with Btrfs           | package   
>   | libbtrfs0             | Library for interacting with Btrfs                              | package  
> ```
> 
> So I suppose the package we're looking for is `libbtrfs-devel `?
> /cc @marcov @GabyCT

Thanks @chavafg  let me have a try with libbtrfs-devel.

**Comment by marcov** [comment]
`libbtrfs-devel` seems the right one, yes

**Comment by lifupan** [comment]
Hi @chavafg, what's the opensuse's default system, a desktop or server? I had tried to reproduce the shimv2 failed issue from my side using a opensuse leap server, but I cannot reproduce this failed. This failure seems related with the apparmor.

**Comment by grahamwhaley** [comment]
Hi @lifupan - @chavafg can confirm later - but having a look around on the Jenkins setup (as I was also curious how we configure this...) I believe the Fedora CI is running in Azure VMs based off the [Fedora Cloud](https://alt.fedoraproject.org/cloud/) images...

**Comment by chavafg** [comment]
I think this is the azure cloud image they provide [1], but there are no details of it in the Azure portal.

[1] https://download.opensuse.org/repositories/Cloud:/Images:/Leap_15.0/images/

**Comment by jodh-intel** [comment]
Hi @lifupan - please can you update the branch as it is conflicted.

**Comment by jodh-intel** [comment]
Hi @lifupan - I'm afraid that the branch went stale whilst we were waiting for another review :cry: Please could you rebase.

Ping @kata-containers/tests !!

**Comment by jodh-intel** [comment]
Hi @lifupan - another PR that has now become conflicted due to lack of reviews. Please can you rebase.

Ping @kata-containers/tests !!

**Comment by jodh-intel** [comment]
The Fedora CI seemed to just give up and go from setup to teardown. I've restarted it to see what happens this time... ;)

The OpenSUSE CI borked on:

```
03:19:27 time="2019-07-31T02:19:27.315954400Z" level=error msg="StopPodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" failed" error="failed to stop container "8e07ef03ab77d633047740b6be9cc915595c170f48cd0f64f76b8cd7bc515436": failed to kill container "8e07ef03ab77d633047740b6be9cc915595c170f48cd0f64f76b8cd7bc515436": unknown error after kill: runc did not terminate sucessfully: container_linux.go:388: signaling init process caused "permission denied"
03:19:27 : unknown" 
03:19:27 time="2019-07-31T02:19:27.316941000Z" level=info msg="RemovePodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446"" 
03:19:27 time="2019-07-31T02:19:27.316983500Z" level=error msg="RemovePodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" failed" error="sandbox container "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" is not fully stopped" 
03:19:27 time="2019-07-31T02:19:27.351188200Z" level=debug msg="received signal" signal=terminated 
```

Looks like an `EPERM` issue?

**Comment by lifupan** [comment]
> The Fedora CI seemed to just give up and go from setup to teardown. I've restarted it to see what happens this time... ;)
> 
> The OpenSUSE CI borked on:
> 
> ```
> 03:19:27 time="2019-07-31T02:19:27.315954400Z" level=error msg="StopPodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" failed" error="failed to stop container "8e07ef03ab77d633047740b6be9cc915595c170f48cd0f64f76b8cd7bc515436": failed to kill container "8e07ef03ab77d633047740b6be9cc915595c170f48cd0f64f76b8cd7bc515436": unknown error after kill: runc did not terminate sucessfully: container_linux.go:388: signaling init process caused "permission denied"
> 03:19:27 : unknown" 
> 03:19:27 time="2019-07-31T02:19:27.316941000Z" level=info msg="RemovePodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446"" 
> 03:19:27 time="2019-07-31T02:19:27.316983500Z" level=error msg="RemovePodSandbox for "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" failed" error="sandbox container "cfc1d688c7e076cea9c6b74f04586d560756d458c225d0ce9d756c1b9f1af446" is not fully stopped" 
> 03:19:27 time="2019-07-31T02:19:27.351188200Z" level=debug msg="received signal" signal=terminated 
> ```
> 
> Looks like an `EPERM` issue?

It seems apparmor blocked this kill, can we stop the apparmor service on opensuse?

**Comment by grahamwhaley** [comment]
Ouch, good diag @lifupan 
/cc @chavafg @GabyCT @marcov

**Comment by marcov** [comment]
Not sure what's exactly failing here.
Stopping AppArmor should be done with `sudo systemctl stop apparmor.service`, assuming this is the proper way to handle the failure.

**Comment by jodh-intel** [comment]
@lifupan, @marcov - I think we need more information here. We *could* just disable Apparmor for testing, but that won't be an option for actual users.

A few questions:

- What Apparmor profiles are active?
- What does Apparmor log when it denies the kill?
- Why don't we see this issue with Ubuntu (which also uses Apparmor)?
- What about the SELinux situation?

**Comment by marcov** [comment]
@jodh-intel, jenkins log are now 404, I am respinning tests now

/test

Also, calling in @saschagrunert for help on cri-o :)

**Comment by saschagrunert** [comment]
> What Apparmor profiles are active?

Hey, for CRI-O we use [this template](https://github.com/containers/libpod/blob/adcde2383f1da517c5788516d639745f5f01d300/pkg/apparmor/apparmor_linux_template.go#L5-L43) to create an apparmor profile on startup. The profile will be only created if the default profile name is configured in `crio.conf` (for 1.15: `apparmor_profile = crio-default-1.15.0`). If the profile name does not match the default one, then we don't add any profile at all because CRI-O expects that profile to be already loaded.

**Comment by marcov** [comment]
@lifupan how have you come to the conclusion that  the issue with opensuse failure is AppArmor  related?

To me it looks like the EPERM error is generated by the agent (libcontainer) inside the VM.

**Comment by lifupan** [comment]
> @lifupan how have you come to the conclusion that the issue with opensuse failure is AppArmor related?
> 
> To me it looks like the EPERM error is generated by the agent (libcontainer) inside the VM.

Hi @marcov , for this failed testcase, it's used runc runtime instead of kata.

**Comment by marcov** [comment]
Thanks for clarifying @lifupan. 
I tried to run the test case locally, with AppArmor enabled, and it does not fail.

However AppArmor is used to limit access to `/proc` and `/sys` inside the container, and I am not aware of any profile on the host limiting what runc can do.
This is the default profile used by containerd: https://github.com/containerd/containerd/blob/master/contrib/apparmor/template.go

**Comment by lifupan** [comment]
> Thanks for clarifying @lifupan.
> I tried to run the test case locally, with AppArmor enabled, and it does not fail.
> 
> However AppArmor is used to limit access to `/proc` and `/sys` inside the container, and I am not aware of any profile on the host limiting what runc can do.
> This is the default profile used by containerd: https://github.com/containerd/containerd/blob/master/contrib/apparmor/template.go

@marcov  actually I'm not sure what's the root cause for this error, and I just suspicious of apparmor, and actually I also cannot reproduce this failure from my side in an opensuse env.

**Comment by lifupan** [comment]
Hi @chavafg , can I get the opensuse vm image used by CI? 
I had tried to reproduce the failure in opensuse, but I  didn't do it using a opensuse system installed from scratch?

File: .ci/setup_env_opensuse.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ declare -A packages=( \
 	[os_tree]="libostree-devel" \
 	[libudev-dev]="libudev-devel" \
 	[metrics_dependencies]="smemstat jq" \
-	[cri-containerd_dependencies]="libseccomp-devel libapparmor-devel make pkg-config" \
+	[cri-containerd_dependencies]="libseccomp-devel libapparmor-devel make pkg-config libbtrfs-devel" \
```
Comment (marcov) [inline .ci/setup_env_opensuse.sh]: @lifupan can you try to add here `apparmor-profiles` and see if it gets better?

Comment (marcov) [reply .ci/setup_env_opensuse.sh]: Or even better `patterns-base-apparmor`

Comment (lifupan) [reply .ci/setup_env_opensuse.sh]: Hi @marcov, thanks, it seems work after I added patterns-base-apparmor.

Comment (marcov) [reply .ci/setup_env_opensuse.sh]: Great, finally we can merge this.

**Pull request merged**